src/System.Net.Primitives/src/System/Net/IPAddressPal.Unix.cs

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Net.Sockets;
+using System.Text;
+
+namespace System.Net
+{
+    internal static class IPAddressPal
+    {
+        public const uint SuccessErrorCode = 0;
+
+        public static unsafe uint Ipv4AddressToString(byte[] address, StringBuilder buffer)
+        {
+            Debug.Assert(address != null);
+            Debug.Assert(address.Length == IPAddressParser.IPv4AddressBytes);
+            Debug.Assert(buffer != null);
+            Debug.Assert(buffer.Capacity >= IPAddressParser.INET_ADDRSTRLEN);
+
+            var sockaddr = new Interop.libc.sockaddr_in {
+                sin_family = Interop.libc.AF_INET,
+                sin_port = 0
+            };
+
+#if BIGENDIAN
+            sockaddr.sin_addr.s_addr = unchecked((uint)(address[3] | (address[2] << 8) | (address[1] << 16) | (address[0] << 24)));
+#else
+            sockaddr.sin_addr.s_addr = unchecked((uint)(address[0] | (address[1] << 8) | (address[2] << 16) | (address[3] << 24)));
+#endif
```

Comment:
I'm also going to file an issue to track revising this: it would be nice to retype the various locations that use network byte order s.t. they're not simple integral values.